src/compiler/crystal/tools/formatter.cr

```diff
@@ -4516,7 +4516,15 @@ module Crystal
         raw_after_comment_value = value[1..-1]
         after_comment_value = raw_after_comment_value.strip
         if after_comment_value.starts_with?("=>")
-          value = "\# => #{after_comment_value[2..-1].strip}"
+          # If a comment starts with `=>`, it indicates the result of an expression which
+          # is a valid expression itself and should be properly formatted.
+          formatted_result_expression = after_comment_value[2..-1]
```

Comment:
Can be replaced by:
`if formatted_result_expression = after_comment_value.lchop? "=>"`

Reply:
To clarify, both the initial if condition and the line with `[2..-1]` can be replaced by the above expression.